docs/reference/concepts/logging.md

```diff
@@ -1,73 +1,146 @@
+{
+    private ILogDispatcher logger;
 
-**Note**: For `LogType.Exception`, add a relevant exception to the `LogEvent` class using the `WithException(Exception)` method. The `WithException` method will be ignored for other `LogType` values.
+    private void OnEnable() 
+    {
+        logger = GetComponent<LinkedEntityComponent>().WorkerSystem.LogDispatcher;
+        logger.HandleLog(...);
+    }
+}
+```
 
-## Creating and using your own dispatcher
+### In the ECS
 
-To create your own dispatcher, make a new class which implements [`ILogDispatcher`]({{urlRoot}}/api/core/i-log-dispatcher):
+You can access the dispatcher through the [`WorkerSystem`]({{urlRoot}}/api/core/worker-system):
 
 ```csharp
-public class MyCustomDispatcher: ILogDispatcher
+public class MySystem : ComponentSystem 
 {
-    public void HandleLog(LogType type, LogEvent logEvent)
+    private ILogDispatcher logger;
+
+    protected override void OnCreate()
     {
-        // Handle Log
+        logger = World.GetExistingSystem<WorkerSystem>().LogDispatcher;
+        logger.HandleLog(...);
     }
+
+    ...
 }
 ```
 
-To use this dispatcher in a specific worker, provide it as an argument when calling `Connect` in the worker connector.
+## Using the log dispatcher
+
+The dispatcher provides a single `HandleLog` function, which takes two arguments:
+
+1. `LogType`, which specifies the verbosity level of the log (e.g. `UnityEngine.LogType.Error`)
+2. `LogEvent`, which stores the message, structured logging data, and the context of the log. See [the API documentation]({{urlRoot}}/api/core/log-event) for usage details.
+
+For example:
+
+```csharp
+logger.HandleLog(LogType.Error, 
+    new LogEvent("Custom error message.")
+        .WithField(LoggingUtils.LoggerName, LoggerName)
+        .WithField(LoggingUtils.EntityId, entityId)
+        .WithField("CustomKey", "CustomValue"));
+```
+
+### The `ForwardingDispatcher`
+
+The `ForwardingDispatcher` converts the Unity `LogType` enum to the SpatialOS `LogLevel` enum using the following table:
+
+| Unity | SpatialOS |
+| --- | --- |
+| `LogType.Exception` | `LogLevel.Error` |
+| `LogType.Error` | `LogLevel.Error` |
+| `LogType.Assert` | `LogLevel.Error` |
+| `LogType.Warning` | `LogLevel.Warning` |
+| `LogType.Log` | `LogLevel.Info` |
+
+> **Note:** By default, messages sent with the [`ForwardingDispatcher`]({{urlRoot}}/api/core/forwarding-dispatcher) with the log level `LogType.Log` are not sent to SpatialOS. You can change this by instantiating the [`ForwardingDispatcher`]({{urlRoot}}/api/core/forwarding-dispatcher) with a different `minimumLogLevel` parameter.
+
+The `ForwardingDispatcher` recognises two special structured logging keys that can be used with the `LogEvent.WithField(string key, object value)` method:
+
+* `LoggingUtils.LoggerName`, which specifies where the log was sent from.
+* `LoggingUtils.EntityId`, which links the log to a specific entity. This lets you filter for a particular entity's logs using the Logger.
```

Comment:
needs a link or description of what the "Logger" is

Reply:
Yeah.. this was elsewhere in the doc, I have no idea why its called this 😕